StRoot/StETofMatchMaker/StETofMatchMaker.cxx

```diff
@@ -3248,77 +3247,61 @@ StETofMatchMaker::sortMatchCases( eTofHitVec inputVec ,  std::map< Int_t, eTofHi
     MMMap.clear();
 
     eTofHitVec ssVec;
-
-     // get multi Hit sets 
-    // int deltaSize = 0;
     
+    // get multi Hit sets 
     eTofHitVecIter tempIter   = tempVec.begin();    
-    eTofHitVecIter erasedIter = erasedVec.begin();
     
-
     if(tempVec.size() < 1 ) return;
+    
+    eTofHitVec storeVecTmp;
+    
+    while(tempVec.size() > 0){
+      
+      std::vector< int > trackIdVec;
+      std::vector< int > hitIdVec;
+      trackIdVec.clear();
+      trackIdVec.resize(0);
+      hitIdVec.clear();
+      hitIdVec.resize(0);
+      tempIter = tempVec.begin();
+      storeVecTmp.push_back(tempVec.at(0));
+      trackIdVec.push_back(tempVec.at(0).trackId);
+      hitIdVec.push_back(tempVec.at(0).index2ETofHit);
+      tempVec.erase(tempVec.begin());
+      bool done = false;
+
+      while(done == false){
```

Comment:
I got into the habit years ago of writing my if statements by placing the value
that I am testing for to the left of the `==` operator.  The reason is because it is too easy to make the mistake of writing `=` instead of `==`. 
```
if ( VALUE == VARIABLE ) { ... } // Compiler happy, Program Correct
if ( VARIABLE = VALUE ) { ... } // Compiler happy, Program Correct
if ( VALUE = VARIABLE ){ ... } // Compiler unhappy, debugging trivial
if ( VARIABLE = VALUE ) { ... } // Compiler happy, debugging hard
```

So prefer `!done` here.

The reason for this is that it is easy to mistakenly write `=` instead of `==`.  If you make that mistake in an if statement, the compiler will let you know

Reply:
> ```
> if ( VALUE = VARIABLE ){ ... } // Compiler unhappy, debugging trivial
> if ( VARIABLE = VALUE ) { ... } // Compiler happy, debugging hard
> ```

Interesting point...I hadn't thought about that `VALUE = VARIABLE` being a straightforward catch by the compiler. Thanks for the tip!